### PR TITLE
Transform userError from response into ApolloError then handle it

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

- Convert success responses with userError into ApolloErrors
- Provide a way to extract error message or provide default
- Show inline error message for SSN field

## Justify why they are needed

- Reduce boilerplate around error handling
- Make sure unhandled errors are reported (they now result in unhandled exceptions)

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

[GRW-2229]: https://hedvig.atlassian.net/browse/GRW-2229?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by alebedev** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1751** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1751" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1751?utm_source=stack-comment).

File: apps/store/src/components/CartInventory/useCampaign.ts [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
 
 type Params = {
   cartId: string
 }
 
 export const useRedeemCampaign = ({ cartId }: Params) => {
-  const getMutationError = useGetMutationError()
-  const [redeemCampaign, result] = useRedeemCampaignMutation()
+  const [redeemCampaign, result] = useRedeemCampaignMutation({
+    onError: () => {}, // Handled via errorMessage
```
Comment (robinandeer) [inline apps/store/src/components/CartInventory/useCampaign.ts:10]: why do we need this one? Will it throw an exception otherwise?

Comment (alebedev) [reply apps/store/src/components/CartInventory/useCampaign.ts:10]: Exactly, it'll throw if error is not handled.  Here we know that error display is taken care of declaratively and callback is not needed

**Comment by alebedev** [comment]
⏳ This pull request is set to merge as part of a Graphite merge job
Job ID: wByc98oIqbaY5M8Mgy0V.
See details on [graphite.dev](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1751)

**Pull request merged**